subsys/rtio/Kconfig

```diff
@@ -8,21 +8,29 @@ if RTIO
 
 config RTIO_SUBMIT_SEM
 	bool "Use a semaphore when waiting for completions in rtio_submit"
+	default n if !MULTITHREADING
+	default y
```

Comment:
I had this initially, but the kernel does something more like what I'm doing here and I was trying to follow along with the same kconfig stuff

https://github.com/zephyrproject-rtos/zephyr/blob/main/kernel/Kconfig#L28

Reply:
The modelling here is wrong. `MULTITHREADING` should be a dependency, not a default condition.
Since this is really a choice between two options, it should be defined as such.
```
choice RTIO_SUMBIT_COMPLETION_METHOD
    prompt "Method to wait until submission has completed"
    default RTIO_SUMBIT_SEM
    default RTIO_SUBMIT_POLL_LOOP

config RTIO_SUMBIT_SEM
    bool "Use sem"
    depends on MULTITHREADING

config RTIO_SUBMIT_POLL_LOOP
    bool "Sit in a loop and poll"

endchoice

```